packages/core/src/xkeys.ts

```diff
@@ -332,19 +349,22 @@ export class XKeys extends EventEmitter {
 	}
 
 	/** Initialize the device. This ensures that the essential information from the device about its state has been received. */
-	public async init(): Promise<void> {
-		const pReceivedVersion = new Promise<void>((resolve) => {
-			this.receivedVersionResolve = resolve
-		})
-		const pReceivedGenerateData = new Promise<void>((resolve) => {
-			this.receivedGenerateDataResolve = resolve
-		})
+	public async init(deviceInfo: DeviceInfo): Promise<void> {
+		if (deviceInfo.productId !== 255) {
```

Comment:
A question for you:
What happens if we try to write to the device? Does it crash? Or nothing?

I'm thinking that if it crashes we might want to guard against writing to it in some other methods as well.